hilti/runtime/src/types/regexp.cc

```diff
@@ -102,15 +102,21 @@ std::tuple<int32_t, stream::View> regexp::MatchState::advance(const stream::View
 
     auto [rc, offset] = _advance(data, data.isFrozen());
 
+    stream::View ndata;
+    if ( offset >= 0 )
+        ndata = data.trim(data.begin() + offset);
+    else
+        ndata = data.trim(data.begin() - static_cast<uint64_t>((-offset)));
```

Comment:
Nit: this is awkward and needed since `SafeConstIterator` implements both `operator-` and `operator+` for `uint64_t` while all we would have need here is `operator+` for `int64_t` (but probably do not want to add since it makes the interface even more complicated). Probably a good idea to at least add a comment so we don't simplify this once we have swapped out the context, e.g.,

```suggestion
    // `SafeConstIterator` implements both `operator+` and `operator-` for `uint64_t`
    // while `offset` is a `int64_t`. Make sure we trim in the correct direction.
    if ( offset >= 0 )
        ndata = data.trim(data.begin() + offset);
    else
        ndata = data.trim(data.begin() - static_cast<uint64_t>((-offset)));
```

Reply:
Yeah, I was thinking about changing the operators, but didn't want to do that as part of this fix to avoid breaking something else.

Reply:
Fixed during merge.